enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+  future enhancement.
+- Details of securing the communication between agents and services to
+  ensure only desired hosts join the cluster are left to a future
+  enhancement.
+- Details of a workflow for deploying in settings with DHCP assignment
+  of IP addresses is left to a future enhancement.
+
+## Proposal
+
+We propose to extend the `openshift-install` binary with a new
+sub-command, `create image` for generating a bootable image containing
+all of the information needed to deploy a cluster.  A command line
+tool avoids issues with prerequisites for deploying a long-running
+service, and can be run on a low-resource host such as a laptop or
+even on a host at a different site before deployment begins.
+
+Most enterprise grade hardware includes a management controller that
+can present an ISO image to a host virtually, allowing the host to
+boot from the image remotely. Hypervisors such as kvm and vSphere also
+support booting from ISOs. Therefore, we will concentrate on that
+format and approach first. Building minimal ISOs with separate root
+filesystems or images that can be used with hosts that PXE boot will
+come later.
+
+We propose to leave up to the user's orchestration tool the
+responsibilities of managing and storing the images, serving them on
+the network, and especially configuring hosts to boot them. This gives
+users the flexibility to perform those operations in any way they
+like, with whatever tool they prefer. A design for an all encompassing
+approach based on tools provided entirely by Red Hat is left for a
+later enhancement.
+
+The `create image` command will need to take inputs similar to what
+the baremetal IPI installer or ZTP APIs use to describe the cluster
+(name, API IPs, etc.) and host-specific network settings (static IPs,
+VLANs, etc.). Using the existing file and custom resource types allows
+us to use the same version management techniques that we do for the
+APIs within a cluster.
+
+The `create image` command will produce a single image as output, to
+save storage space and reduce management complexity. The assisted
+installer service can already generate images with boot scripts that
+recognize which host is booting and apply host-specific network
+settings and roles (such as running different services) accordingly,
+and we will reuse that technique (and code).
+
+One host, designated as "node0", will be configured to run the
+assisted service and a client that uses the assisted service API to
+automate the deployment.
+
+All other hosts will run the assisted agent, configured to talk to the
+service on node0.
+
+When all of the images are booted, the client on node0 will use the
+assisted service to trigger the deployment of the cluster (including
+applicable pre-flight validations) without further interaction from
+the user.
+
+The `create image` command will produce as output credentials to
+access the cluster that is about to be created. This allows a client,
```

Comment:
This is the advice from the docs.
```
It is recommended that you use Ignition config files within 12 hours after they are generated because the 24-hour certificate rotates from 16 to 22 hours after the cluster is installed. By using the Ignition config files within 12 hours, you can avoid installation failure if the certificate update runs during installation.
```

Reply:
the final certs are generated by assisted-service calling openshift-installer in node0. In this stage, only the admin key and certificate should be the final ones. So, while more will be generated, the main task should be to find a way to set the CA generated in the create image step to be the same one that will be used to regenerate certs in the node0 stage.